Week 3 Tutorial Step 3

nif33 opened the pull request:
Add comment section with JSON conversion of data.
Also remove an unused variable (enumeration object) for the slideshow

nif33 committed:
add a fetch function greeting - week3step2

nif33 committed:
Merge branch 'master' of https://github.com/nif33/my-portfolio into JenniferDevelopment-week3
update current working branch

nif33 committed:
week 3 step 3

nif33 committed:
remove greeting-container

nif33 committed:
add gson to pom.xml file

nif33 committed:
move comment box creation to a function

nif33 committed:
separate comment boxes

nif33 committed:
removed enum from slideshow (is not used)

juanalonsocruz commented on portfolio/src/main/webapp/script.js:
```diff
@@ -28,21 +28,35 @@ function addRandomFact() {
+function newCommentBox(comment){
+  const commentBox = document.createElement("p");
+  commentBox.id = "comment"
+  commentBox.innerText = comment;
```
https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes

juanalonsocruz commented on portfolio/src/main/webapp/script.js:
```diff
@@ -28,21 +28,35 @@ function addRandomFact() {
+function loadComments() {
+  fetch('/data').then(response => response.json()).then((comments) => {
+    for(const comment of comments) {
+      newCommentBox(comment);
```
Nit: This is more of a personal opinion. I prefer top level DOM modifications to be in top level methods.

It would look like this:
```
for(const comment of comments) {
  document.getElementById('comment-container').appendChild(newCommentBox(comment));
}
```

If you don't like this, then another suggestion would be to change the function name `newCommentBox` to `appendNewCommentBox'

juanalonsocruz reviewed (approved):
Two small comments. Otherwise, approving.

ajpaon commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -24,9 +27,29 @@
+  private List<String> comments;
+
+  @Override
+  public void init() {
```
No need to specialize init for this class. This will be run for every request, which isn't necessary because comments always contains the same entries. Instead you can make comments a static class member and only initialize it once.

nif33 replied:
I am not too sure on what this means, but I changed comments to a static variable, and initialized it in doGet instead of an init() function?

ajpaon commented on portfolio/src/main/webapp/script.js:
```diff
@@ -28,21 +28,35 @@ function addRandomFact() {
+ */
+function newCommentBox(comment){
+  const commentBox = document.createElement("p");
+  commentBox.id = "comment"
```
Do you need to give this element an explicit ID? This means every comment box will have the same ID, which is probably not what you want.

nif33 committed:
move DOM modification to top level methods

appending new comment box to comment-container was in newCommentBox function, now in loadComments function

nif33 committed:
assign comment to a class instead of id

newCommentBox() assigns the class 'comment' to each comment instead of the id 'comment'
CSS modified to alter a class instead of an id

nif33 committed:
remove init function and change comments var to static type

ajpaon commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -24,9 +27,27 @@
+    comments = new ArrayList<>();
+    comments.add("Hi there!");
+    comments.add("How are you?");
+    comments.add("Howdy");
```
Sorry for being unclear before. I was trying to point out that, since the comments list never changes, we don't need to re-initialize it on every request. Functions like doGet and init are called for every request, so we should do the initialization outside of them. You can initialize static variables in the same line as the declaration (you might find Arrays.asList [1] useful)

[1] https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#asList-T...-

nif33 committed:
initialize Arraylist outside of functions
